server/lib/device/device.get.js

```diff
@@ -44,13 +44,37 @@ async function get(options) {
     order: [[optionsWithDefault.order_by, optionsWithDefault.order_dir]],
   };
 
+  // fix attributes
+  if (optionsWithDefault.attributes_device && optionsWithDefault.attributes_device.length > 0) {
+    queryParams.attributes = optionsWithDefault.attributes_device;
+  }
+  if (optionsWithDefault.attributes_device_feature && optionsWithDefault.attributes_device_feature.length > 0) {
+    queryParams.include[0].attributes = optionsWithDefault.attributes_device_feature;
+  }
+  if (optionsWithDefault.attributes_device_param && optionsWithDefault.attributes_device_param.length > 0) {
+    queryParams.include[1].attributes = optionsWithDefault.attributes_device_param;
+  }
+  if (optionsWithDefault.attributes_device_room && optionsWithDefault.attributes_device_room.length > 0) {
+    queryParams.include[2].attributes = optionsWithDefault.attributes_device_room;
+  }
+  if (optionsWithDefault.attributes_device_service && optionsWithDefault.attributes_device_service.length > 0) {
+    queryParams.include[3].attributes = optionsWithDefault.attributes_device_service;
+  }
+
```

Comment:
What is this and why is it useful?

Reply:
this part is for reduce data up from database (reduce attribute in select query) so it's just to improve the performance of object serialization .